File: samples/python/hello_ai.py
import os
from openai import AzureOpenAI

# 環境変数から大切な情報を取得します
api_key = os.getenv('AZURE_OPENAI_API_KEY')
endpoint = os.getenv('AZURE_OPENAI_ENDPOINT')

if not api_key or not endpoint:
    print("環境変数 AZURE_OPENAI_API_KEY と AZURE_OPENAI_ENDPOINT を設定してください")
    exit(1)

# AIクライアントを準備します
client = AzureOpenAI(
    api_key=api_key,
    api_version="2024-02-15-preview",
    azure_endpoint=endpoint
)

# AIと会話します
response = client.chat.completions.create(
    model="gpt-4o-mini",
    messages=[
        {"role": "user", "content": "こんにちは！私の名前は田中です。"}
    ]
)

# AIの返事を表示します
print("AI:", response.choices[0].message.content)

File: samples/nodejs/hello_ai.js
import { AzureOpenAI } from "openai";

// 環境変数から大切な情報を取得します
const endpoint = process.env.AZURE_OPENAI_ENDPOINT;
const apiKey = process.env.AZURE_OPENAI_API_KEY;

if (!endpoint || !apiKey) {
  console.error("環境変数 AZURE_OPENAI_API_KEY と AZURE_OPENAI_ENDPOINT を設定してください");
  process.exit(1);
}

// AIクライアントを準備します
const client = new AzureOpenAI({
  endpoint: endpoint,
  apiKey: apiKey,
  deployment: "gpt-4o-mini",
  apiVersion: "2024-12-01-preview",
});

// AIと会話する関数を作ります
async function chatWithAI() {
  // AIと会話します
  const result = await client.chat.completions.create({
    model: "gpt-4o-mini",
    messages: [{ role: "user", content: "何か面白いことを言ってください" }],
  });

  // AIの返事を表示します
  console.log("AI:", result.choices[0].message.content);
}

// プログラムを実行します
chatWithAI().catch(console.error);
